compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -1346,10 +1349,27 @@ impl NodeInterner {
         let struct_type = constructor_expression.r#type.borrow();
         let field_names = struct_type.field_names();
 
-        match field_names.iter().find(|field_name| field_name.0 == expr_rhs.0) {
-            Some(found) => Some(Location::new(found.span(), struct_type.location.file)),
-            None => None,
-        }
+        field_names.iter().find(|field_name| field_name.0 == expr_rhs.0).map(|found_field_name| {
+            Location::new(found_field_name.span(), struct_type.location.file)
+        })
+    }
+
+    fn handle_struct_member_access_for_self(
```

Comment:
The contents of this function make sense but the naming confuses me. It sounds like it is supposed to be resolving member accesses to the `self` parameter only but the contents should work for any expression. We should probably do this type based resolution by default and remove the other method of checking if the variable was defined as the result of a constructor expression.